### PR TITLE
ENH Adds no_batch_dim to FractionalMaxPool2d

### PR DESCRIPTION
**Pull request by thomasjpfan**
Towards #60585

### CODE REVIEW

File: aten/src/ATen/native/cuda/FractionalMaxPool2d.cu [hunk 1]
```diff
@@ -241,7 +241,7 @@ void fractional_max_pool2d_backward_out_cuda_template(
             gradInput_.size(0));
   dim3 block(outputPlaneSize > 128 ? 128 : outputPlaneSize);
 
-  auto devIndices = indices.packed_accessor<int64_t, 4>();
+  auto devIndices = indices_.packed_accessor<int64_t, 4>();
```
Comment (thomasjpfan) [inline aten/src/ATen/native/cuda/FractionalMaxPool2d.cu:244]: Bug fix where the no_batch_dim tests is a non regression test.

File: torch/nn/functional.py [hunk 2]
```diff
@@ -458,10 +458,14 @@ def fractional_max_pool2d_with_indices(
...
-        output_size = [int(input.size(2) * _output_ratio[0]), int(input.size(3) * _output_ratio[1])]
+        output_size = [int(input.size(-2) * _output_ratio[0]), int(input.size(-1) * _output_ratio[1])]
 
     if _random_samples is None:
-        _random_samples = torch.rand(input.size(0), input.size(1), 2, dtype=input.dtype, device=input.device)
+        n_batch = input.size(0)
+        if input.dim() == 3:
+            # no_batch dimension
+            n_batch = 1
+        _random_samples = torch.rand(n_batch, input.size(-3), 2, dtype=input.dtype, device=input.device)
```
Comment (thomasjpfan) [inline torch/nn/functional.py:468]: I'm open to making `_random_samples.dim() == 3`.

File: torch/testing/_internal/common_nn.py [hunk 3]
```diff
@@ -1214,6 +1214,56 @@ def fractional_max_pool2d_test(test_case):
             fullname='FractionalMaxPool2d_size')
 
 
+def fractional_max_pool2d_no_batch_dim_test(test_case, use_random_samples):
```
Comment (jbschlosser) [inline torch/testing/_internal/common_nn.py:1217]: Awesome test coverage! :)

File: torch/nn/functional.py (hunk 2 above)
Comment (jbschlosser) [reply torch/nn/functional.py:465]: Probably the way to go for the cleanest solution

**Review by jbschlosser — COMMENTED** [review]
Looks great! I like your idea of having `_random_samples` accept no-batch-dims as well

**Review by jbschlosser — COMMENTED** [review]
As we discussed, should investigate whether we need to revert the `_random_samples` change to get the older tests passing again

**Comment by thomasjpfan** [comment]
The existing `test_fractional_max_pool2d` test explicitly checks for no batch dimensions and sets `random_samples` to have a batch dimension for both cases. Because of this, I have went back to expecting a 4D randomSample.

**Review by jbschlosser — APPROVED** [review]
LGTM!

**Pull request closed without merging**